File: interface/src/main/java/cn/itcast/core/service/GoodsService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.entity.GoodsEntity;
import cn.itcast.core.pojo.entity.PageResult;
import cn.itcast.core.pojo.good.Goods;
import cn.itcast.core.pojo.item.Item;
import org.springframework.stereotype.Repository;

import java.util.List;


public interface GoodsService {
    //添加商品
    void add(GoodsEntity goodsEntity);

    //分页加条件查询
    PageResult search(Integer page, Integer rows, Goods goods);

    //点修改回显数据
    GoodsEntity findOne(Long id);

    //修改
    void update(GoodsEntity goodsEntity);

    //删除(逻辑删除,并不是真的在数据库中删除,只是在页面上不显示)
    void delete(Long[] ids);

    //上架
    void upShelf(Long[] ids);

    //下架
    void downShelf(Long[] ids);

    //上架功能查询商品列表
    public List<Goods> findGoodsForUpShelf(String sellerId);

    //下架功能查询商品列表
    public List<Goods> findGoodsForDownShelf(String sellerId);


    //批量修改状态
    void updateStatus(Long[] ids, String status);



}

File: web-shop/src/main/webapp/js/service/goodsService.js
//服务层
app.service('goodsService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../goods/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../goods/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../goods/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../goods/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../goods/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../goods/delete.do?ids='+ids);
	}
    //上架功能查询商品列表
    this.findGoodsForUpShelf=function(){
        return $http.get('../goods/findGoodsForUpShelf.do');
    }
    //上架
    this.upShelf=function(ids){
        return $http.get('../goods/upShelf.do?ids='+ids);
    }
    //下架功能查询商品列表
    this.findGoodsForDownShelf=function(){
        return $http.get('../goods/findGoodsForDownShelf.do');
    }
    //下架
    this.downShelf=function(ids){
        return $http.get('../goods/downShelf.do?ids='+ids);
    }
    //提交审核
    this.updateStat=function(ids){
        return $http.get('../goods/updateStat.do?ids='+ids);
    }
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../goods/search.do?page='+page+"&rows="+rows, searchEntity);
	}


});
